### ISSUE TITLE
Update S2583: Should not raise issues for boolean constants and literals in conditions

### ISSUE THREAD

**Issue by jccollet** [SonarSource/sonar-csharp#804]
The following exceptions have been added to RSPEC-2583:
This rule will not raise an issue in either of these cases:

- When the condition is a single `const bool`
```
const bool debug = false;
//...
if (debug)
{
  // Print something
}

```
- When the condition is the literal `true` or `false`.

In these cases it is obvious the code is as intended.
